src/core/currency/wallet/currency-wallet-api.ts

```diff
@@ -693,7 +693,9 @@ export function combineTxWithFile(
     networkFee: tx.networkFee.get(tokenId) ?? '0',
     otherParams: { ...tx.otherParams },
     ourReceiveAddresses: tx.ourReceiveAddresses,
-    parentNetworkFee: tx.networkFee.get(null) ?? '0',
+    ...(walletCurrency === currencyCode
+      ? {}
+      : { parentNetworkFee: tx.networkFee.get(null) ?? '0' }),
```

Comment:
I would prefer `parentNetworkFee: walletCurrency === currencyCode ? undefined : tx.networkFee.get(null) ?? '0'`.

There is no need to remove the field itself - setting it to `undefined` is simpler.